passthrough-server/src/main/java/org/terracotta/passthrough/PassthroughServiceRegistry.java

```diff
@@ -47,19 +47,27 @@ public PassthroughServiceRegistry(String entityClassName, String entityName, lon
     this.entityName = entityName;
     this.consumerID = consumerID;
     
+    Map<Class<?>, ServiceProvider> overrideServiceTypes = new HashMap<Class<?>, ServiceProvider>();
+    for (ServiceProvider overrideServiceProvider : overrideServiceProviders) {
+      for (Class<?> serviceType : overrideServiceProvider.getProvidedServiceTypes()) {
+        overrideServiceTypes.put(serviceType, overrideServiceProvider);
+      }
+    }
+
     Map<Class<?>, ServiceProvider> tempProviders = new HashMap<Class<?>, ServiceProvider>();
     for(ServiceProvider provider : serviceProviders) {
       for (Class<?> serviceType : provider.getProvidedServiceTypes()) {
         // We currently have no way of handling multiple providers.
-        ServiceProvider existingProviderForType = tempProviders.get(serviceType);
-        if (existingProviderForType != null && overrideServiceProviders.contains(provider)) {
-          // Since user provided override service providers are always last in the ordered list of service providers,
-          // this means that this is an user provided service that is allowed to override
-          tempProviders.remove(serviceType);
+        if(overrideServiceTypes.containsKey(serviceType)) {
+          //install override ServiceProvider
+          if(!tempProviders.containsKey(serviceType)) {
```

Comment:
My question is regarding which provider will be left in the map.  It looks like this change will leave the previous entry, not the override.